app/flatpak-builtins-build-commit-from.c

```diff
@@ -450,12 +454,44 @@ flatpak_builtin_build_commit_from (int argc, char **argv, GCancellable *cancella
 
       dst_collection_id = ostree_repo_get_collection_id (dst_repo);
 
+      collection_ids = g_ptr_array_new_with_free_func (g_free);
+      if (dst_collection_id)
+        {
+          main_collection_id = dst_collection_id;
+          g_ptr_array_add (collection_ids, g_strdup (dst_collection_id));
+        }
+
+      if (opt_extra_collection_ids != NULL)
+        {
+          for (j = 0; opt_extra_collection_ids[j] != NULL; j++)
+            {
+              const char *cid = opt_extra_collection_ids[j];
+              if (main_collection_id == NULL)
+                main_collection_id = cid; /* Fall back to first arg */
+
+              if (g_strcmp0 (cid, dst_collection_id) != 0)
+                g_ptr_array_add (collection_ids, g_strdup (cid));
+            }
+        }
+
+      g_ptr_array_sort (collection_ids, (GCompareFunc)flatpak_strcmp0_ptr);
+
       /* Copy old metadata */
       g_variant_builder_init (&metadata_builder, G_VARIANT_TYPE ("a{sv}"));
 
       /* Bindings. xa.ref is deprecated but added anyway for backwards compatibility. */
       g_variant_builder_add (&metadata_builder, "{sv}", "ostree.collection-binding",
-                             g_variant_new_string (dst_collection_id ? dst_collection_id : ""));
+                             g_variant_new_string (main_collection_id ? main_collection_id : ""));
+      if (collection_ids->len > 0)
+        {
+          g_autoptr(GVariantBuilder) cr_builder = g_variant_builder_new (G_VARIANT_TYPE ("a(ss)"));
+
+          for (j = 0; j < collection_ids->len; j++)
+            g_variant_builder_add (cr_builder, "(ss)", g_ptr_array_index (collection_ids, j), dst_ref);
+
+          g_variant_builder_add (&metadata_builder, "{sv}", "ostree.collection-refs-binding",
```

Comment:
Post-hoc review (sorry for the delay): As per my comments in https://github.com/ostreedev/ostree/pull/1805#pullrequestreview-196962466, I think it would be useful if flatpak (and OSTree) guaranteed to insert the bindings into the metadata in alphabetical order. It makes future searches/comparisons easier, and means commit metadata is more reproducible, regardless of the order of command line options to `ostree commit` or `flatpak build-commit-from`.